cdap-app-fabric/src/main/java/co/cask/cdap/metadata/MetadataHttpHandler.java

```diff
@@ -113,7 +112,7 @@ public void getProgramMetadata(HttpRequest request, HttpResponder responder,
                                  @PathParam("program-type") String programType,
                                  @PathParam("program-id") String programId,
                                  @QueryParam("scope") String scope) throws NotFoundException, BadRequestException {
-    ProgramId program = new ProgramId(namespaceId, appId, ProgramType.valueOf(programType), programId);
+    ProgramId program = new ProgramId(namespaceId, appId, programType, programId);
```

Comment:
It does, but what is actually passed here for `programType` isn't correct and `ProgramType.valueOfCategoryName` should be used instead.